x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java

```diff
@@ -217,44 +231,269 @@ public ConnectorSyncJob(StreamInput in) throws IOException {
-            builder.field(CANCELED_AT_FIELD.getPreferredName(), canceledAt);
-            builder.field(COMPLETED_AT_FIELD.getPreferredName(), completedAt);
+            if (cancelationRequestedAt != null) {
+                builder.field(CANCELATION_REQUESTED_AT_FIELD.getPreferredName(), cancelationRequestedAt);
+            }
+            if (canceledAt != null) {
+                builder.field(CANCELED_AT_FIELD.getPreferredName(), canceledAt);
+            }
+            if (completedAt != null) {
+                builder.field(COMPLETED_AT_FIELD.getPreferredName(), completedAt);
+            }
 
             builder.startObject(CONNECTOR_FIELD.getPreferredName());
             {
                 builder.field(Connector.ID_FIELD.getPreferredName(), connector.getConnectorId());
-                builder.field(Connector.FILTERING_FIELD.getPreferredName(), connector.getFiltering());
-                builder.field(Connector.INDEX_NAME_FIELD.getPreferredName(), connector.getIndexName());
-                builder.field(Connector.LANGUAGE_FIELD.getPreferredName(), connector.getLanguage());
-                builder.field(Connector.PIPELINE_FIELD.getPreferredName(), connector.getPipeline());
-                builder.field(Connector.SERVICE_TYPE_FIELD.getPreferredName(), connector.getServiceType());
-                builder.field(Connector.CONFIGURATION_FIELD.getPreferredName(), connector.getConfiguration());
+                if (connector.getFiltering() != null) {
+                    builder.field(Connector.FILTERING_FIELD.getPreferredName(), connector.getFiltering());
+                }
+                if (connector.getIndexName() != null) {
+                    builder.field(Connector.INDEX_NAME_FIELD.getPreferredName(), connector.getIndexName());
+                }
+                if (connector.getLanguage() != null) {
+                    builder.field(Connector.LANGUAGE_FIELD.getPreferredName(), connector.getLanguage());
+                }
+                if (connector.getPipeline() != null) {
+                    builder.field(Connector.PIPELINE_FIELD.getPreferredName(), connector.getPipeline());
+                }
+                if (connector.getServiceType() != null) {
+                    builder.field(Connector.SERVICE_TYPE_FIELD.getPreferredName(), connector.getServiceType());
+                }
+                if (connector.getConfiguration() != null) {
+                    builder.field(Connector.CONFIGURATION_FIELD.getPreferredName(), connector.getConfiguration());
+                }
             }
             builder.endObject();
 
             builder.field(CREATED_AT_FIELD.getPreferredName(), createdAt);
-            builder.field(DELETED_DOCUMENT_COUNT.getPreferredName(), deletedDocumentCount);
-            builder.field(ERROR_FIELD.getPreferredName(), error);
+            builder.field(DELETED_DOCUMENT_COUNT_FIELD.getPreferredName(), deletedDocumentCount);
+            if (error != null) {
```

Comment:
Since error can be `null` in the index, if someone sets it to `null` via API we should still write it to make it backward compatible

Reply:
The current API spec forces a non-null error:

```
PUT _connector/_sync_job/<id>/_error
{
  message: string;
}

```,

which is probably wrong, if we want to unset the error. We should discuss for `8.13`, whether unsetting an error is/shoud possible and which status it should go to, if you do so.